// sails-hook-mailer/index.js
module.exports = function nodemailerHook(sails) {

    /*
    function sendTestMail(){
        var nodemailer = require('nodemailer');

        var smtpConfig = {
            host: 'smtp.gmail.com',
            port: 587,//465,
            secure: false,//true // use SSL
            auth: {
                user: 'adamsawicki89@gmail.com',
                pass: 'pass'
            },
            tls: {
                rejectUnauthorized: false
            }
        };

        var transporter = nodemailer.createTransport(smtpConfig);

        // setup e-mail data with unicode symbols
        var mailOptions = {
            from: '"Fred Foo ?" <foo@blurdybloop.com>', // sender address
            to: 'adamsawicki89@gmail.com', // list of receivers
            subject: 'Hello ?', // Subject line
            text: 'Hello world ?', // plaintext body
            html: '<b>Hello world ?</b>' // html body
        };

        // send mail with defined transport object
        transporter.sendMail(mailOptions, function(error, info){
            if(error){
                return console.log(error);
            }
            console.log('Message sent: ' + info.response);
        });
    };*/

    function sendMail(smtpConfig, mailOptions, callback){

        var nodemailer = require('nodemailer');
        var transporter = nodemailer.createTransport(smtpConfig);

        // send mail with defined transport object
        transporter.sendMail(mailOptions, function(error, info){
            /*
            if(error){
                return console.log(error);
            }
            console.log('Message sent: ' + info.response);*/

            if(callback && typeof callback === "function"){
                callback(error, info);
            }

        });
    }

    return {
        sendMail:sendMail
    }
};
